boards/nordic/nrf52840dk/src/main.rs

```diff
@@ -419,6 +421,38 @@ pub unsafe fn reset_handler() {
     //         components::multi_alarm_test_component_buf!(nrf52840::rtc::Rtc),
     //     );
 
+    //--------------------------------------------------------------------------
+    // USB CTAP EXAMPLE
+    //--------------------------------------------------------------------------
+    // Uncomment to experiment with this.
+
+    // // Create the strings we include in the USB descriptor.
+    // let strings = static_init!(
+    //     [&str; 3],
+    //     [
+    //         "Nordic Semiconductor", // Manufacturer
+    //         "nRF52840dk - TockOS",  // Product
+    //         "serial0001",           // Serial number
+    //     ]
+    // );
+
+    // let ctap_send_buffer = static_init!([u8; 64], [0; 64]);
+    // let ctap_recv_buffer = static_init!([u8; 64], [0; 64]);
+
+    // let (ctap, _ctap_driver) = components::ctap::CtapComponent::new(
+    //     &nrf52840::usbd::USBD,
+    //     0x1915, // Nordic Semiconductor
+    //     0x503a, // lowRISC generic FS USB
+    //     strings,
+    //     board_kernel,
+    //     ctap_send_buffer,
+    //     ctap_recv_buffer,
+    // )
+    // .finalize(components::usb_ctap_component_helper!(nrf52840::usbd::Usbd));
+
+    // ctap.enable();
+    // ctap.attach();
```

Comment:
From what I tested, this is the part that currently doesn't work on nRF52840-DK, as the USB is generally not ready yet this early in the board initialization process just after reboot.

Reply:
Do you want to send a patch that changes this to something that does work for the nRF52840-DK?

Doing this works fine for OpenTitan. @bradjc added this after testing it on a board, I don't have one to test on.

Reply:
I don't think it's a reasonable goal for this PR to fix the usb lifecycle. It's a bigger patch, at least on the nordic chip, that needs to happen separately.
Unless someone is interested taking it, as the main users of this board, that's a problem that the OpenSK team will have to work on.